parlai/core/torch_generator_agent.py

```diff
@@ -894,15 +907,22 @@ def eval_step(self, batch):
         if self.rank_candidates:
             cand_choices, cand_scores = self.rank_eval_label_candidates(batch, bsz)
 
-        text = [self._v2t(p) for p in preds] if preds is not None else None
+        text = [self._v2t(pred_data[0]) for pred_data in beam_preds_scores] if beam_preds_scores is not None else None
+
+        if self.output_token_losses:
+            generated_text_token_info = [list(zip([self.dict[int(token)] for token in pred_data[0]],pred_data[2].tolist(), pred_data[3].tolist())) for pred_data in beam_preds_scores] if beam_preds_scores is not None else None
```

Comment:
this is far too hard to read. Let's just be explicit in the loop and avoid zip madness